src/lib/containers/widgets/query-filter/QueryFilter.tsx

```diff
@@ -138,22 +139,36 @@ export const applyChangesToRangeColumn = (
 export const QueryFilter: React.FunctionComponent<QueryFilterProps> = ({
   facetsToFilter,
 }): JSX.Element => {
+  const [facets, setFacets] = React.useState<FacetColumnResult[]>([])
+  const [facetFiltersShown, setFacetFiltersShown] = React.useState<string[]>(
+    facetsToFilter ?? [],
+  )
+  const { facetAliases, topLevelControlsState } = useQueryVisualizationContext()
+  const { showFacetFilter } = topLevelControlsState
   const { data, isLoadingNewBundle, getLastQueryRequest, executeQueryRequest } =
     useQueryContext()
 
-  const { facetAliases, topLevelControlsState } = useQueryVisualizationContext()
+  React.useEffect(() => {
+    if (data) {
+      const facets = data.facets as FacetColumnResult[]
+      if (facetFiltersShown.length > 0) {
+        let newArr = facets.filter(facet => {
+          return facetsToFilter
+            ? facetFiltersShown.includes(facet.columnName)
+            : !facetFiltersShown.includes(facet.columnName)
+        })
+        setFacets(newArr)
+      } else {
+        setFacets(facets)
```

Comment:
When the component is mounted, the state isn't always updated which leads to not showing the facets occasionally. Once any of the facet chips are clicked it does render. 
<img width="257" alt="Screen Shot 2022-09-13 at 9 58 25 AM" src="https://user-images.githubusercontent.com/60489495/189962027-ee5cff7c-4612-478e-8130-55867b2f07fa.png">